### PR TITLE
feat: add environment variable to explicitly set custom access key ID and secret access key for AWS

### PR DESCRIPTION
**Pull request by lukeramsden**
This should solve #169

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -6867,7 +6878,7 @@ packages:
     peerDependencies:
       tailwindcss: '>=2.0.0 || >=3.0.0 || >=3.0.0-alpha.1'
     dependencies:
-      tailwindcss: 3.2.4_ts-node@10.9.1
+      tailwindcss: 3.2.4_v776zzvn44o7tpgzieipaairwm
```
Comment (robinpellegrims) [inline pnpm-lock.yaml:6881]: Just wondering why the tailwindcss dependencies changed in the lockfile, maybe we could revert anything not related to @aws-sdk/types in this lockfile?

File: libs/nx-remotecache-s3/src/lib/s3-client.ts [hunk 2]
```diff
@@ -5,20 +5,35 @@ import { getDefaultRoleAssumerWithWebIdentity } from '@aws-sdk/client-sts';
...
 
 export const buildS3Client = (
   options: Pick<S3ClientConfig, 'endpoint' | 'region' | 'forcePathStyle'> &
     Pick<DefaultProviderInit, 'profile'>
 ) => {
-  const provider = defaultProvider({
-    profile: getEnv(ENV_PROFILE) ?? options.profile,
-    roleAssumerWithWebIdentity: getDefaultRoleAssumerWithWebIdentity(),
-  });
+  const awsAccessKeyIdOverride = getEnv(ENV_ACCESS_KEY_ID);
```
Comment (robinpellegrims) [inline libs/nx-remotecache-s3/src/lib/s3-client.ts:21]: Personally I would extract this logic related to building the credentials provider into a separate function. This way we can keep it simple in here and just have a const variable `const provider = buildCredentials();`

**Review by robinpellegrims — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good to me, just a few minor thoughts.

File: pnpm-lock.yaml (hunk 1 above)
Comment (lukeramsden) [reply pnpm-lock.yaml:6881]: Yeah it's a bit weird - I'll go through and revert any that don't make sense

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -21498,13 +21509,15 @@ packages:
       tslib: 2.5.0
     dev: true
 
-  /tailwindcss/3.2.4_ts-node@10.9.1:
+  /tailwindcss/3.2.4_v776zzvn44o7tpgzieipaairwm:
```
Comment (robinpellegrims) [inline pnpm-lock.yaml:21512]: Should we remove this change and the postcss one as well?

**Review by robinpellegrims — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml (hunk 3 above)
Comment (lukeramsden) [reply pnpm-lock.yaml:21512]: Oops - not sure how I missed that one. Very big file 😁

**Review by robinpellegrims — APPROVED** [review]

File: pnpm-lock.yaml (hunk 3 above)
Comment (robinpellegrims) [reply pnpm-lock.yaml]: No worries :-)

**Pull request merged**